### DISCUSSION TITLE
Enable PDF preview with Avo field file(s)

### DISCUSSION THREAD

**Discussion by lpe** [avo-hq/avo#1497] [Help] [answered]
I know how to do it in ROR but what would be the best strategy with Avo : tool, custom firlf, eject, wrapper ... ? Thx by adance for your help.

**Comment by adrianthedev** [comment, marked as answer]
Hey @lpe. It depends on where you want to show it. 

If that PDF is a column in the database, the best way would probably be a custom PDF field, or if you want more space to display it in, a custom tool.

Another strategy might be to duplicate (or extend and push in a PR) the file/files field to support that. The preview can be a small preview in the gallery (for the files field) and on click could open up a modal showing a bigger version.

![CleanShot 2022-12-24 at 14 17 02@2x](https://user-images.githubusercontent.com/1334409/209435972-d215c193-5858-4970-ac4e-7a79a2311574.jpg)

**Reply by lpe** [reply]
Thank you for the answer. I am not yet experienced enough to propose a PR ;-) I actually want to propose a preview like for an image and open the PDF when you click on it. I will try your suggestions to extend the file/files field.

**Reply by adrianthedev** [reply]
Got it!

Please let me know if you get stuck anywhere, and maybe I can chime in and help you get unstuck.

Here are a couple of resources to help you:
 - file field [here](https://github.com/avo-hq/avo/blob/main/lib/avo/fields/file_field.rb). the files field is next to it. You might want to create a new `ExtendedFileField < Avo::FileField` class.
 - [the directory with the view components](https://github.com/avo-hq/avo/tree/main/app/components/avo/fields/file_field). You might want to duplicate these inside your app for your new field to add the preview.
 - documentation for custom fields [here](https://docs.avohq.io/2.0/custom-fields.html) where I explain how you can create custom fields.
